azurerm/helpers/validate/kubernetes.go

```diff
@@ -0,0 +1,40 @@
+package validate
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func KubernetesAdminUserName() schema.SchemaValidateFunc {
+	return validation.StringMatch(
+		regexp.MustCompile(`^[A-Za-z][-A-Za-z0-9_]*$`),
+		"AdminUserName must start with alphabet and/or continue with alphanumeric characters, underscores, hyphens.")
+}
+
+func KubernetesCidr() schema.SchemaValidateFunc {
```

Comment:
This should be added to `helpers/validate/network.go` as it is not kubernetes specific